Clarify breaking release frequency (#7934)
bevyengine/bevy: README.md

## WARNING

> Bevy is still in the _very_ early stages of development. APIs can and will change (now is the time to make suggestions!). Important features are missing. Documentation is sparse. Please don't build any serious projects in Bevy unless you are prepared to be broken by API changes on a roughly quarterly basis.

Reviewer: The official train release schedule is every 3 months, so I would rephrase it with this in mind. https://bevyengine.org/news/bevy-0-6/#the-train-release-schedule

Personally, I still think the user on hacker news assumed something that was never said. Bevy uses semver and nothing here claims otherwise but I guess it's true that being more accurate doesn't hurt this warning.

Author: That link says "approximately once every three months", this says "roughly quarterly". Those would be the same, right?

> Personally, I still think the user on hacker news assumed something that was never said

Agreed on this point

Reviewer 2: Yep, those are the same period of time.

Reviewer: > Those would be the same, right?

Correct, what I meant is that I think this should explicitly mention the concept of train release and maybe even link it.

> Bevy is still in the early stages of development. Important features are missing. Documentation is sparse. And minor versions of Bevy containing breaking changes to the API are released [approximately once every 3 months](https://bevyengine.org/news/bevy-0-6/#the-train-release-schedule). Use at your own risk.

Reviewer 2: ```suggestion
Bevy is still in the early stages of development. Important features are missing. Documentation is sparse. A new version of Bevy containing serious breaking changes to the API is released [approximately once every 3 months](https://bevyengine.org/news/bevy-0-6/#the-train-release-schedule). Use at your own risk.
```

Reviewer 2: They are technically minor versions, but I think that carries the wrong connotation.

> Bevy is still in the early stages of development. Important features are missing. Documentation is sparse. A new version of Bevy containing serious breaking changes to the API is released [approximately once every 3 months](https://bevyengine.org/news/bevy-0-6/#the-train-release-schedule). Use at your own risk.

Reviewer 3: ```suggestion
Bevy is still in the early stages of development. Important features are missing. Documentation is sparse. A new version of Bevy containing breaking changes to the API is released [approximately once every 3 months](https://bevyengine.org/news/bevy-0-6/#the-train-release-schedule). We provide [migration guides](https://bevyengine.org/learn/book/migration-guides/), but we can't guarantee migrations will always be easy. Use only if you are willing to work in this environment.
```

"Use at your own risk" seems a bit to harsh and ominous to me. The kind of thing you sign on a waiver before doing something hazardous to your health. I swapped that out for something a bit less loaded. Also added a link to migration guides.

Reviewer 2: I'm on board with that wording.
